### PR TITLE
Add data source for service principals

### PR DESCRIPTION
**Pull request by rondefreitas**
- return `application_id` from `databricks_current_user` if that user is a Service Principal
- add `databricks_service_principal` data source (maybe could enhance later with a display name lookup, but for now, you need the application id)
- add tests for data source
- update documentation

Hopefully this isn't too shabby... first time writing anything in Go.

### CODE REVIEW

File: docs/data-sources/current_user.md [hunk 1]
```diff
@@ -56,6 +56,7 @@ output "job_url" {
 Data source exposes the following attributes:
 
 * `id` -  The id of the calling user.
+* `application_id` - Application ID of the [service principal](../resources/service_principal.md) if the currently logged-in user is a service principal, e.g. `813073e7-3b08-4c0d-8619-39bcfa4ac632`
```
Comment (nfx) [inline docs/data-sources/current_user.md:59]: can you anonymise `813073e7-3b08-4c0d-8619-39bcfa4ac632`?

File: docs/data-sources/service_principal.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
+---
+subcategory: "Security"
+---
+
+# databricks_service_principal Data Source
+
+-> **Note** If you have a fully automated setup with workspaces created by [databricks_mws_workspaces](../resources/mws_workspaces.md) or [azurerm_databricks_workspace](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/databricks_workspace), please make sure to add [depends_on attribute](../index.md#data-resources-and-authentication-is-not-configured-errors) in order to prevent _authentication is not configured for provider_ errors.
+
+Retrieves information about [databricks_service_principal](../resources/user.md).
```
Comment (nfx) [inline docs/data-sources/service_principal.md:9]: `user.md` -> `service_principal.md`

File: scim/data_service_principal.go [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+// DataSourceUser returns information about user specified by user name
+func DataSourceServicePrincipal() *schema.Resource {
+	return &schema.Resource{
```
Comment (nfx) [inline scim/data_service_principal.go:14]: please rewrite with the new data resource framework: https://github.com/databrickslabs/terraform-provider-databricks/blob/master/jobs/data_jobs.go

**Review by nfx — CHANGES_REQUESTED** [review]

File: docs/data-sources/service_principal.md [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+## Example Usage
+
+Adding service principal "813073e7-3b08-4c0d-8619-39bcfa4ac632" to administrative group
```
[the remark this answers is not in the document]
Comment (rondefreitas) [reply docs/data-sources/service_principal.md:13]: will do @nfx . That was actually randomly generated for the docs, but I'll update accordingly.

**Comment by nfx** [comment]
@rondefreitas any update on review comments?

**Comment by nfx** [comment]
@rondefreitas are you still interested in finishing this pull request?

**Comment by nfx** [comment]
@rondefreitas are you still interested in finishing this pull request?

**Pull request closed without merging**